executor/aggfuncs/func_avg.go

```diff
@@ -56,7 +58,19 @@ func (e *baseAvgDecimal) AppendFinalResult2Chunk(sctx sessionctx.Context, pr Par
 	finalResult := new(types.MyDecimal)
 	err := types.DecimalDiv(&p.sum, decimalCount, finalResult, types.DivFracIncr)
 	if err != nil {
-		return errors.Trace(err)
+		return err
+	}
+	// Make the decimal be the result of type inferring.
+	frac := e.args[0].GetType().Decimal
```

Comment:
Why not directly use the result type of the aggregate function, but use the result type of the argument instead?

Reply:
@zz-jason 
We could not get the result type of function here except for adding an interface,
the argument type has been wrap with cast to make sure it's the same with the function.